airflow/datasets/__init__.py

```diff
@@ -182,6 +190,28 @@ def __eq__(self, other: Any) -> bool:
     def __hash__(self) -> int:
         return hash(self.uri)
 
+    @property
+    def normalized_uri(self) -> str | None:
```

Comment:
Do we intend to use this anywhere else? This logic can simply live in the OL provider, from what I can tell.

Reply:
I wrote it in the provider at first and it's not used anywhere else yet, but it relates closely to a Dataset class and it uses only "core" mechanisms related to the URI so i thought this is a suitable place for it and it makes a lot of sense ot keep it closer to the Dataset class. Does it make sense?